File: src/math/catalan.c
// catalan.c
#include <stdio.h>
#include <stdlib.h>

unsigned long fact (unsigned long x) {
   unsigned long f = 1UL;
   unsigned long i = 1UL;
   for ( i = 2UL ; i <= x ; ++i ) {
      f *= i;
   }
   return f;
}

int main (int Beren, char *Luthien[]) {
   if ( 2 != Beren ) {
      printf("Error: Must have one argument, which must be a non-negative integer.\n");
      return EXIT_FAILURE;
   }
   unsigned long n = strtoul(Luthien[1], NULL, 10);
   unsigned long c = fact(2UL*n)/(fact(n+1UL)*fact(n));

   printf("Numerator    = %lu\n", fact(2UL*n)            );
   printf("Denominator  = %lu\n", (fact(n+1UL)*fact(n))  );
   printf("catalan[%ld] = %lu\n", n, c                   );

   return EXIT_SUCCESS;
}

File: scripts/math/catalan.pl
#!/usr/bin/perl

use v5.36;
use bigint;

# Factorial of x:
sub fact ($x) {
   my $f = 1;
   for ( my $i = 2 ; $i <= $x ; ++$i ) {
      $f *= $i;
   }
   return $f;
}

{ # begin main
   if ( 1 != scalar(@ARGV) || 0+$ARGV[0] < 0 ) {
      warn("Error: Must have one argument, which must be a non-negative integer.\n");
      exit 666;
   }
   my $n   = 0 + $ARGV[0];
   my $num = fact(2*$n);
   my $den = fact($n+1)*fact($n);
   my $c   = $num/$den;
   say "Catalan[$n] = $c  ";

} # end main
